xds/internal/xdsclient/xdsresource/unmarshal_cds_test.go

```diff
@@ -1620,3 +1622,181 @@ func (s) TestUnmarshalCluster(t *testing.T) {
+	envconfig.XDSOutlierDetection = true
+	defer func() { envconfig.XDSOutlierDetection = oldOutlierDetectionSupportEnv }()
+	odProto := func(od *v3clusterpb.OutlierDetection) *v3clusterpb.Cluster {
+		// Cluster parsing doesn't fail with respect to fields orthogonal to
+		// outlier detection.
+		return &v3clusterpb.Cluster{
+			Name:                 clusterName,
+			ClusterDiscoveryType: &v3clusterpb.Cluster_Type{Type: v3clusterpb.Cluster_EDS},
+			EdsClusterConfig: &v3clusterpb.Cluster_EdsClusterConfig{
+				EdsConfig: &v3corepb.ConfigSource{
+					ConfigSourceSpecifier: &v3corepb.ConfigSource_Ads{
+						Ads: &v3corepb.AggregatedConfigSource{},
+					},
+				},
+			},
+			LbPolicy:         v3clusterpb.Cluster_ROUND_ROBIN,
+			OutlierDetection: od,
+		}
+	}
+	odUpdate := func(od *OutlierDetection) ClusterUpdate {
+		return ClusterUpdate{
+			ClusterName:      clusterName,
+			LRSServerConfig:  ClusterLRSOff,
+			OutlierDetection: od,
+		}
+	}
+
+	odDefaultUpdate := &OutlierDetection{
+		Interval:                       10 * time.Second,
+		BaseEjectionTime:               30 * time.Second,
+		MaxEjectionTime:                300 * time.Second,
+		MaxEjectionPercent:             10,
+		SuccessRateStdevFactor:         1900,
+		EnforcingSuccessRate:           100,
+		SuccessRateMinimumHosts:        5,
+		SuccessRateRequestVolume:       100,
+		FailurePercentageThreshold:     85,
+		EnforcingFailurePercentage:     0,
+		FailurePercentageMinimumHosts:  5,
+		FailurePercentageRequestVolume: 50,
+	}
+
+	odProtoWithValidFields := &v3clusterpb.OutlierDetection{
+		Interval:                       &durationpb.Duration{Seconds: 1},
+		BaseEjectionTime:               &durationpb.Duration{Seconds: 2},
+		MaxEjectionTime:                &durationpb.Duration{Seconds: 3},
+		MaxEjectionPercent:             &wrapperspb.UInt32Value{Value: 1},
+		SuccessRateStdevFactor:         &wrapperspb.UInt32Value{Value: 2},
+		EnforcingSuccessRate:           &wrapperspb.UInt32Value{Value: 3},
+		SuccessRateMinimumHosts:        &wrapperspb.UInt32Value{Value: 4},
+		SuccessRateRequestVolume:       &wrapperspb.UInt32Value{Value: 5},
+		FailurePercentageThreshold:     &wrapperspb.UInt32Value{Value: 6},
+		EnforcingFailurePercentage:     &wrapperspb.UInt32Value{Value: 7},
+		FailurePercentageMinimumHosts:  &wrapperspb.UInt32Value{Value: 8},
+		FailurePercentageRequestVolume: &wrapperspb.UInt32Value{Value: 9},
+	}
+	odUpdateWithValidFields := &OutlierDetection{
+		Interval:                       time.Second,
+		BaseEjectionTime:               time.Second * 2,
+		MaxEjectionTime:                time.Second * 3,
+		MaxEjectionPercent:             1,
+		SuccessRateStdevFactor:         2,
+		EnforcingSuccessRate:           3,
+		SuccessRateMinimumHosts:        4,
+		SuccessRateRequestVolume:       5,
+		FailurePercentageThreshold:     6,
+		EnforcingFailurePercentage:     7,
+		FailurePercentageMinimumHosts:  8,
+		FailurePercentageRequestVolume: 9,
+	}
```

Comment:
Since these vars are only used by one test case, I would rather inline them there instead of defining them here.

Reply:
Sure. Since defaultUpdate was also only used by one test case, I inlined that one as well.